diff --git a/ac.scm b/ac.scm
index b16531f..24ddb67 100644
--- a/ac.scm
+++ b/ac.scm
@@ -69,6 +69,7 @@
         ; this line could be removed without changing semantics
         ((eq? (xcar (xcar s)) 'compose) (ac (decompose (cdar s) (cdr s)) env))
         ((pair? s) (ac-call (car s) (cdr s) env))
+        ((eof-object? s) (exit))
         (#t (err "Bad object in expression" s))))
 
 (define (literal? x)
