diff --git a/arc.arc b/arc.arc
index c2b6805..1113969 100644
--- a/arc.arc
+++ b/arc.arc
@@ -757,6 +757,9 @@
       (w/stdout ,gv ,@body)
       (inside ,gv))))
 
+(mac toerr body
+  `(w/stdout (stderr) ,@body))
+
 (mac fromstring (str . body)
   (w/uniq gv
    `(w/instring ,gv ,str
@@ -1358,7 +1361,7 @@
   `(let ,var (table) ,@body ,var))
 
 (def ero args
-  (w/stdout (stderr) 
+  (toerr
     (each a args 
       (write a)
       (writec #\space))
